api/list.go

```diff
@@ -125,19 +128,35 @@ func (c *Client) ListObjects(ctx context.Context, list runtimeclient.ObjectList,
 		return fmt.Errorf("error when searching for projects: %w", err)
 	}
 
+	projectsSize := len(projects)
+	var wg sync.WaitGroup
+	wg.Add(projectsSize)
+	ch := make(chan reflect.Value, projectsSize)
+
 	for _, proj := range projects {
 		tempOpts := slices.Clone(opts.clientListOptions)
-		// we ensured the list is a pointer type and that is has an
-		// 'Items' field which is a slice above, so we don't need to do
-		// this again here and instead use the reflect functions directly.
-		tempList := reflect.New(reflect.TypeOf(list).Elem()).Interface().(runtimeclient.ObjectList)
-		tempList.GetObjectKind().SetGroupVersionKind(list.GetObjectKind().GroupVersionKind())
-		if err := c.List(ctx, tempList, append(tempOpts, runtimeclient.InNamespace(proj.Name))...); err != nil {
-			return fmt.Errorf("error when searching in project %s: %w", proj.Name, err)
-		}
-		tempListItems := reflect.ValueOf(tempList).Elem().FieldByName("Items")
-		for i := 0; i < tempListItems.Len(); i++ {
-			items.Set(reflect.Append(items, tempListItems.Index(i)))
+		go func() {
+			defer wg.Done()
+			// we ensured the list is a pointer type and that is has an
+			// 'Items' field which is a slice above, so we don't need to do
+			// this again here and instead use the reflect functions directly.
+			tempList := reflect.New(reflect.TypeOf(list).Elem()).Interface().(runtimeclient.ObjectList)
+			tempList.GetObjectKind().SetGroupVersionKind(list.GetObjectKind().GroupVersionKind())
+			if err := c.List(ctx, tempList, append(tempOpts, runtimeclient.InNamespace(proj.Name))...); err != nil {
+				log.Printf("error when searching in project %s: %w", proj.Name, err)
```

Comment:
I guess just logging the error is good enough

Reply:
I would use our existing printing function here `format.PrintWarningf` as `log.Printf` won't really be consistent with other nctl output.

Reply:
Fixed.